src/averageAges.js

```diff
@@ -1,61 +1,63 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+const getSex = (people, sex) => people.filter(person => person.sex === sex);
+
+/* Calculate Men Average Age */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let isMen = getSex(people, 'm');
+
+  const isMenInCentury = isMen.filter(person =>
+    Math.ceil(person.died / 100) === century);
+
+  isMen = century
+    ? isMenInCentury
+    : isMen;
+
+  const averageAgeMen = isMen.reduce(
+    (sum, man) => sum + (man.died - man.born), 0
+  ) / isMen.length;
+
+  return averageAgeMen;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
+/* Calculate Women Average Age */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let isWoman = getSex(people, 'f');
+
+  const womenWithKids = people.filter(
+    person => people.some(child => child.mother === person.name));
+
+  isWoman = withChildren
+    ? womenWithKids
+    : isWoman;
+
+  const averageAgeWomen = isWoman.reduce(
+    (sum, woman) => sum + (woman.died - woman.born), 0
+  ) / isWoman.length;
+
+  return averageAgeWomen;
 }
 
-/**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
+/* Calculate Men Average Difference */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let children = people.filter(
+    child => people.some(mother => mother.name === child.mother));
+
+  const isBoy = getSex(children, 'm');
+
+  children = onlyWithSon
+    ? isBoy
+    : children;
+
+  const averageAgeDiff = children.reduce(
+    (sum, child) => sum + (child.born - people.find(
+      mom => child.mother === mom.name).born), 0
+  ) / children.length;
```

Comment:
that's unreadable, I'd something like that
```suggestion
  const averageAgeDiff = children.reduce((sum, child) => {
  	const mother = people.find(person => child.mother === person.name);
  	const ... = sum + (child.born - mother.born);
  	
  	return ...;
  }, 0) / children.length;
```